sharktank_models/test_suite/ireers_tools/fixtures.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import pytest
+
+from typing import Dict, Sequence, Union
+from pathlib import Path
+import subprocess
+import time
+import os
+
+from .artifacts import (
+    Artifact,
+    FetchedArtifact,
+    ProducedArtifact,
+)
+
+
+IREE_COMPILE_QOL_FLAGS = [
+    "--mlir-timing",
+    "--mlir-timing-display=list",
+    "--iree-consteval-jit-debug",
+]
+
+def fetch_source_fixture(url: str, *, group: str):
+    art = FetchedArtifact(url=url, group=group)
+    art.start()
+    return art
+
+
+
+def iree_compile(source: Artifact, flags: Sequence[str], vmfb_path: Path):
+    if not os.path.exists(vmfb_path.parent):
+        os.makedirs(vmfb_path.parent)
+    sep = "\n  "
+    print("**************************************************************")
+    print(f"  {sep.join(flags)}")
+    exec_args = (
+        [
+            "iree-compile",
+            "-o",
+            str(vmfb_path),
+            str(source.path),
+        ]
+        + IREE_COMPILE_QOL_FLAGS
+        + flags
+    )
+    print("Exec:", " ".join(exec_args))
+    start_time = time.time()
+    subprocess.run(exec_args, check=True, capture_output=True, cwd=vmfb_path.parent)
```

Comment:
Duplicate code with `run_iree_command` in `sharktank_models/test_suite/benchmarks/run_benchmarks_for_model.py`? We have lots of versions of this in the repository now, would prefer to only fork again if truly needed.

Same for `iree_run_module` and `iree_benchmark_module` in this file.